### PR TITLE
String Sum Testing.

### PR DESCRIPTION
**Pull request by AlexanderVladimirov9090**
Made new String Summator and test it.

### CODE REVIEW

File: src/com/clouway/stringsum/ArithmeticTest.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    calculator.add("");
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testAddNewLineThrowsNullPointerException() {
+    double result = calculator.add("\n");
+  }
+
+  @Test(expected = NullPointerException.class)
+  public void testAddOnlyTextThrowsNullPointerException() {
```
Comment (mgenov) [inline src/com/clouway/stringsum/ArithmeticTest.java]: junit 4 don't require method names to start with test 

```java
// Test annotation already shows that this method is a test method
@Test 
public void addOnlyTextThrowsNullPointerException() {
   calculator.add("asdfasdf");
}
```

File: src/com/clouway/stringsum/ArithmeticTest.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+  @Test
+  public void testAddNumbersEndTextReturnSumOfNumbers() {
+    double result = calculator.add("1 2 3 asd");
+    assertEquals(6, result, 0);
+  }
+
+  @Test
+  public void testAddTextNewLinesNumbersReturnSumOfNumbers() {
+    double result = calculator.add("import java.util.regex.Pattern;\n" +
```
Comment (mgenov) [inline src/com/clouway/stringsum/ArithmeticTest.java]: Using of  "aaa\n34\nbbb4ccc\n11" will be more then enough. This snippet looks a really big and doesn't  give much value.

File: src/com/clouway/stringsum/Demo.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package com.clouway.stringsum;
+
+/**
+ * Created by clouway on 15.06.16.
+ *
+ * @author alexandervladimirov1902@gmail.com
+ *         (Alexander Vladimirov)
+ */
+public class Demo {
```
Comment (mgenov) [inline src/com/clouway/stringsum/Demo.java]: Demo classes are no longer required as Tests are now written.

File: src/com/clouway/stringsum/test/ArithmeticTest.java [hunk 4]
```diff
@@ -0,0 +1,115 @@
+package com.clouway.stringsum;
```
Comment (mgenov) [inline src/com/clouway/stringsum/test/ArithmeticTest.java]: Tests should not be placed in the SRC folder. 

src/com/clouway/stringsum/Arithmetic
test/com/clouway/stringsum/ArithmeticTest

File: test/ArithmeticTest.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
+import com.clouway.stringsum.Arithmetic;
```
Comment (mgenov) [inline test/ArithmeticTest.java]: Not placed in the same package.

**Comment by AlexanderVladimirov9090** [comment]
Now it is done.

File: src/test/ArithmeticTest.java [hunk 6]
```diff
@@ -0,0 +1,79 @@
+package test;
```
Comment (mgenov) [inline src/test/ArithmeticTest.java]: The convention says that it should be placed in the same package as code it tests.

**Comment by mgenov** [comment]
LGTM !

**Pull request merged**